ADApp/ADSrc/asynNDArrayDriver.cpp

```diff
@@ -430,8 +510,9 @@ asynStatus asynNDArrayDriver::writeOctet(asynUser *pasynUser, const char *value,
     /* Set the parameter in the parameter library. */
     status = (asynStatus)setStringParam(addr, function, (char *)value);
 
-    if (function == NDAttributesFile) {
-        this->readNDAttributesFile(value);
+    if ((function == NDAttributesFile) ||
+        (function == NDAttributesMacros)) {
+        this->readNDAttributesFile();
```

Comment:
Minor observation: on IOC loading, autosave will write sequentially to `NDAttributesFile` and then to `NDAttributesMacros`, causing the file to be loaded and parsed twice (which can take a few hundred of ms each time, from what I observed). A minor workaround would be to reverse the order of the records in `NDArrayBase.template` so that `NDAttributesMacros` is restored first but won't trigger a file reading/parsing since `NDAttributesFile` will be empty at that time. 

Ideally there would be some mechanism to set both parameters at the same time, but having something like `NDAttributesFileProcess` would be a bit cumbersome and break previous expectations.